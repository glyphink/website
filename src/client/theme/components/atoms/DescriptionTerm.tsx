/**
 * @file Components - DescriptionTerm
 * @module website/theme/components/atoms/DescriptionTerm
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type FRC,
  type JSX
} from 'react'
import type DescriptionTermProps from './DescriptionTerm.props.mts'

/**
 * A term in a description or definition list.
 *
 * > ðŸ‘‰ **Note**: Must be used inside a `DescriptionList` component.
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/dt
 * @see {@linkcode DescriptionTermProps}
 * @see {@linkcode FRC}
 *
 * @category
 *  atoms
 *
 * @const {FRC<DescriptionTermProps>} DescriptionTerm
 */
const DescriptionTerm: FRC<DescriptionTermProps> = forwardRef<
  HTMLElement,
  DescriptionTermProps
>(
  /**
   * @param {Props<DescriptionTermProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<dt>` element
   */
  (
    props: Props<DescriptionTermProps>,
    ref: ForwardedRef<HTMLElement>
  ): JSX.Element => {
    return <dt {...props} ref={ref} />
  }
)

DescriptionTerm.displayName = 'DescriptionTerm'

export default DescriptionTerm
