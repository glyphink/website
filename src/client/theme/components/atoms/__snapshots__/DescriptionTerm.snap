// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/DescriptionTerm > should render <dt> 1`] = `
<dt>
  Beast of Bodmin
</dt>
`;
